### PR TITLE
remove /pkg/storage dependency on pkg/api/errors

### PR DESCRIPTION
**Pull request by adohe-zz**
Refer #23284 for more details. @caesarxuchao @lavalamp ptal.

### CODE REVIEW

**Comment by hongchaodeng** [comment]
Take a look at https://github.com/kubernetes/kubernetes/pull/23521

It seemed like the errors definition part is already done. You just need to change the watch_cache part.

File: pkg/storage/watch_cache_test.go [hunk 1]
```diff
@@ -130,7 +129,7 @@ func TestEvents(t *testing.T) {
 		if err == nil {
 			t.Errorf("expected error too old")
 		}
-		if _, ok := err.(*errors.StatusError); !ok {
+		if !IsGoneError(err) {
 			t.Errorf("expected error to be of type StatusError")
```
Comment (lavalamp) [inline pkg/storage/watch_cache_test.go]: Update error message?

File: pkg/storage/errors.go [hunk 2]
```diff
@@ -104,3 +108,44 @@ func isErrCode(err error, code int) bool {
 	}
 	return false
 }
+
+// InvalidError is generated when an error caused by invalid API object
+// occurs in the storage package.
+type InvalidError struct {
```
Comment (lavalamp) [inline pkg/storage/errors.go]: Hm, I thought we already had this.

Comment (caesarxuchao) [inline pkg/storage/errors.go]: It's added in #22965. Could you rebase? Thanks.

Comment (AdoHe) [inline pkg/storage/errors.go]: of course, will rebase.

**Comment by AdoHe** [comment]
@lavalamp I just update this, ptal.

**Comment by lavalamp** [comment]
You still need to add to InterpretWatchError: https://github.com/kubernetes/kubernetes/blob/c6e995a824094a96f7d43a25e897283f83a12997/pkg/api/errors/storage/storage.go#L100

**Comment by AdoHe** [comment]
oh, sure.

**Comment by AdoHe** [comment]
Jenkins GCE Node e2e failed because of this: `/tmp/hudson7822584213509620883.sh: line 4: cd: go/src/k8s.io/kubernetes: No such file or directory`
Jenkins GCE e2e failed because of this test flake:
```
[Fail] [k8s.io] ServiceAccounts [It] should mount an API token into pods [Conformance] 
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/util.go:1692
```
Jenkins unit/integration failed because of this:
```
errors in package "k8s.io/kubernetes/pkg/storage/etcd":
import k8s.io/kubernetes/pkg/util/net did not match any allowed prefix

errors in package "k8s.io/kubernetes/pkg/storage":
import k8s.io/kubernetes/pkg/conversion did not match any allowed prefix

errors in package "k8s.io/kubernetes/pkg/storage/testing":
import k8s.io/kubernetes/pkg/api/v1 did not match any allowed prefix

errors in package "k8s.io/kubernetes/pkg/storage/etcd3":
import k8s.io/kubernetes/pkg/conversion/queryparams did not match any allowed prefix
```
@lavalamp could you help me what I should do to fix the GCE Node e2e issue, also it seems that I have to add some allowed prefix to `/pkg/storage/.import-restrictions`, simply add `ForbiddenPrefixes` is not enough.

**Comment by lavalamp** [comment]
https://github.com/kubernetes/kubernetes/blob/c6e995a824094a96f7d43a25e897283f83a12997/cmd/libs/go2idl/import-boss/main.go#L23

You can add just `/` as an allowed prefix.

**Comment by lavalamp** [comment]
The node thing is a flake that's affecting everyone-- if you change your import restriction file and re-push it should resolve itself.

File: pkg/storage/.import-restrictions [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+  "Rules": [
+    {
+      "SelectorRegexp": "k8s[.]io",
+      "AllowedPrefixes": [
+        "/"
```
Comment (lavalamp) [inline pkg/storage/.import-restrictions]: Sorry for talking nonsense--use "" and see if that works.

**Comment by jfrazelle** [comment]
this should have changed the imports at the top of the file if its really removing the dependency on the errors package

**Comment by lavalamp** [comment]
Closing due to inactivity but please reopen when you get it working!

**Pull request closed without merging**